blockstack/blockstack-browser
# Name registration errors and deny button

### What this does
This PR fixes two main issues:

## Fix id.blockstack name registration https://github.com/blockstack/blockstack-browser/issues/1764
- create id / sign in here -> https://deploy-preview-1898--reporter-beaver-73821.netlify.com
- go to IDs page
- hit 'more'
- generate a new ID
- hit 'add username' 
- search for a free subdomain
- when you are on the page that asks if you'd like to register for free, [**block requests to registrar.blockstack.org**](https://developers.google.com/web/updates/2017/04/devtools-release-notes#block-requests)
- attempt to register the subdomain
- you will see a failure message (this message will not display a support email, however if the server responds with 409, it will show `support@blockstack.org`).
## Reduce misuse of "Deny" button during onboarding https://github.com/blockstack/blockstack-browser/issues/1816
- how to test same tab example:
  - Login / create an account here -> https://deploy-preview-1898--reporter-beaver-73821.netlify.com
  - use this app to sign in https://react-blockstack-redux-bundler-yjnojbyuwr.now.sh/sign-in
  - notice that the 'Cancel' button is in the top right
  - hit the cancel button
- how to test native browser version:
  - clone this repo and checkout this PR `feature/better-errors-notifications`
  - `yarn && yarn dev`
  - set dev mode on the blockstack browser app (opt click icon in nav bar)
  - auth with any blockstack app
  - hit cancel in auth modal
  - window should close

For mobile apps, I've opted to hide the cancel button in general as per @yknl's suggestion.
